src/plugins/navigation_embeddable/public/editor/open_editor_flyout.tsx

```diff
@@ -35,14 +40,45 @@ const NavigationEmbeddablePanelEditor = withSuspense(
  * @throws in case user cancels
  */
 export async function openEditorFlyout(
-  initialInput?: Omit<NavigationEmbeddableInput, 'id'>,
+  initialInput: NavigationEmbeddableInput,
   parentDashboard?: DashboardContainer
 ): Promise<Partial<NavigationEmbeddableInput>> {
+  const attributeService = getNavigationEmbeddableAttributeService();
+  const { attributes } = await attributeService.unwrapAttributes(initialInput);
+  const isByReference = attributeService.inputIsRefType(initialInput);
+
   return new Promise((resolve, reject) => {
     const closed$ = new Subject<true>();
 
-    const onSave = (partialInput: Partial<NavigationEmbeddableInput>) => {
-      resolve(partialInput);
+    const onSaveToLibrary = async (newLinks: NavigationEmbeddableLink[]) => {
```

Comment:
When the user cancels out of the save modal, the `isSaving` state in the `NavigationEmbeddablePanelEditor` component remains `true` and never gets set to `false`: 

https://github.com/elastic/kibana/assets/8698078/e88c19ab-057c-4854-8ceb-f0f8f6897461

Reply:
I refactored this to use a better save modal and the `isSaving` state should return to `false` after canceling the modal.